### ISSUE TITLE
Difficulty Reproducing Results in four_lane Task

### ISSUE THREAD

**Issue by xingzhe-cpu** [ucd-dare/CarDreamer#25]
Hello,  

I carefully followed your installation steps and ran the script as instructed:  

```bash
bash train_dm3.sh 2000 0 --task carla_four_lane --dreamerv3.logdir ./logdir/carla_four_lane
```

However, I was unable to reproduce your results in the `four_lane` scenario. The outputs I obtained seem to differ from those shown in your paper/documentation. Could you please provide some insights into what might be causing this discrepancy?  

Here are some screenshots of my results(black curve) and the results in the paper(blue curve) for reference:  

<img src="https://github.com/user-attachments/assets/b62ce2b9-3b7e-415c-9ae8-f347fd9df4cc" width="300">
<img src="https://github.com/user-attachments/assets/94840676-057d-461f-a91c-32b1cba1b4ca" width="300">  

I would greatly appreciate any guidance you can provide.  

Thank you for your time and help!

**Comment by gaodechen** [comment]
Hi!

Thank you for the post!

We didn't have such issues across all the training sessions on the `four_lane` task including training sessions with different observability.

* Wondering if any of the yaml files (`common.yaml`, `tasks.yaml`, `dreamerv3.yaml`) was changed before running the training scripts, or if you got any messages in the log file.

* The curve you showed was `sum of abs rewards` and the paper reports the `score` curve in the tensorboard, though the sum abs reward should be similar in late training.

* Feel free to check out the policy rollout animation within tensorboard and see how it goes during training process.

<img width="906" alt="Image" src="https://github.com/user-attachments/assets/98f15548-2da0-4317-92e4-e2f5eab0b391" />

* Another way to troubleshoot is to re-run the training sessions from scratch (avoid using the checkpoint in the failed session), and see it gets rid of sub-optimality.

Feel free to drop us emails (mine is dcgao at ucdavis edu) and we can discuss via email or wechat.

**Comment by xingzhe-cpu** [comment]
Thank you for your response! I plan to rerun the experiment on another remote server in our lab. For any further questions regarding the experiment, I will reach out to you via email.  

I appreciate your support!

**Comment by xingzhe-cpu** [comment]
Thank you for your response! I plan to rerun the experiment on another remote server in our lab. For any further questions regarding the experiment, I will reach out to you via email.  

I appreciate your support!

**Comment by gaodechen** [comment]
> Thank you for your response! I plan to rerun the experiment on another remote server in our lab. For any further questions regarding the experiment, I will reach out to you via email.
> 
> I appreciate your support!

Hi Xingzhe, would you pull the latest version? I just checked the recent commits and the reason might be a change in `common.yaml`. Our previous experiments in the paper used discrete action spaces. So I rollbacked the config and it should be good now! Thank you for the catch!

**Comment by xingzhe-cpu** [comment]
Thank you for your patient！I will have a try.

**Comment by xingzhe-cpu** [comment]
After modifying `commo.yaml` to use a discrete action space, I was indeed able to successfully reproduce the results. Please see the output below:

<p float="left"> <img src="https://github.com/user-attachments/assets/bbd130d5-3e5f-47ba-88b0-6ef47b3de4c5" width="300" /> <img src="https://github.com/user-attachments/assets/c510cad8-ae19-4e93-8a4b-153e732fdb94" width="300" /> </p>
Additionally, could you kindly provide a full example of how to run `eval_dm3.sh`  with all the parameters?
